x-pack/test/cloud_security_posture_functional/pages/rules.ts

```diff
@@ -92,8 +92,12 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
       });
 
       it('Clicking the integrations counter button leads to the integration page', async () => {
-        await rule.rulePage.clickIntegrationsEvaluatedButton();
-        await pageObjects.common.waitUntilUrlIncludes('add-integration/kspm');
+        await retry.waitFor('Wait until url includes', async () =>
+          rule.rulePage.doesElementContainsUrl(
+            'rules-counters-integrations-evaluated-button',
+            'add-integration/kspm'
+          )
+        );
```

Comment:
The tests says it check that after clicking a button it leads to the integration page.
but the test itself checks that the button element contains the href attribute and returns a boolean.

perhaps after the waiting, we should add previous checks as well?

```
await rule.rulePage.clickIntegrationsEvaluatedButton();
await pageObjects.common.waitUntilUrlIncludes('add-integration/kspm');
```

Also, consider renaming the waitFor text to something like "wait until button's href includes url"

Reply:
The issue with this step is that, the test seems to be problematic when we run it on ci (as in it the button never gets clicked) as such, I think it might be better to do it this way (by checking the URL instead of actually clicking on it)